### ISSUE TITLE
AIO Nextcloud 29.0.2 Cadviewer doesn't work with error E127 

### ISSUE THREAD

**Issue by lehakatana** [CADViewer/NextCloud#92]
Fresh installation AIO nextcloud, i tried install Cadviewer via store and from git, result same. 

![image](https://github.com/CADViewer/NextCloud/assets/42839959/80ccf288-4d95-46f0-b49c-523677837b85)
Can't find anything about error E127

My AIO behind haproxy,
Debug said that i have wrong .htaccess, but I did everything according to the guide.
content after ### in .htaccess:

#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####

ErrorDocument 403 /index.php/error/403
ErrorDocument 404 /index.php/error/404
<IfModule mod_rewrite.c>
  Options -MultiViews
  RewriteRule ^core/js/oc.js$ index.php [PT,E=PATH_INFO:$1]
  RewriteRule ^core/preview.png$ index.php [PT,E=PATH_INFO:$1]
  RewriteCond %{REQUEST_FILENAME} !\.(css|js|mjs|svg|gif|png|html|ttf|woff2?|ico|jpg|jpeg|map|webm|mp4|mp3|ogg|wav|flac|wasm|tflite)$
  RewriteCond %{REQUEST_FILENAME} !/core/ajax/update\.php
  RewriteCond %{REQUEST_FILENAME} !/core/img/(favicon\.ico|manifest\.json)$
  RewriteCond %{REQUEST_FILENAME} !/(cron|public|remote|status)\.php
  RewriteCond %{REQUEST_FILENAME} !/ocs/v(1|2)\.php
  RewriteCond %{REQUEST_FILENAME} !/robots\.txt
  RewriteCond %{REQUEST_FILENAME} !/(ocs-provider|updater)/
  RewriteCond %{REQUEST_URI} !^/\.well-known/(acme-challenge|pki-validation)/.*
  RewriteCond %{REQUEST_FILENAME} !/richdocumentscode(_arm64)?/proxy.php$
  RewriteCond %{REQUEST_FILENAME} !/apps/cadviewer/converter/php/*\.*
  RewriteRule . index.php [PT,E=PATH_INFO:$1]
  RewriteBase /
  <IfModule mod_env.c>
    SetEnv front_controller_active true
    <IfModule mod_dir.c>
      DirectorySlash off
    </IfModule>
  </IfModule>
</IfModule>

i pinned log and full htaccess file
[call-Api_Conversion_log .txt](https://github.com/user-attachments/files/15970054/call-Api_Conversion_log.txt)
[htaccess.txt](https://github.com/user-attachments/files/15970058/htaccess.txt)

I have classic nextcloud, all works fine after installation and set permissions. 
But in AIO have problems, help pls!

When i pressed flush cache get error:
![image](https://github.com/CADViewer/NextCloud/assets/42839959/7ebd4946-7dda-4a00-931a-fff2894de9de)

i found command line in Api_conversation_log:
$command_line: "/var/www/html/apps/cadviewer/converter//converters/ax2024/linux//ax2023_L64_23_12_130" -i="/var/www/html/apps/cadviewer/converter//converters/files/f1483319627.dwg" -o="/var/www/html/apps/cadviewer/converter//converters/files/f1483319627.svgz"  -f="svgz" -model -extents -fpath="/var/www/html/apps/cadviewer/converter//converters/ax2024/linux/fonts/"  -lpath="/var/www/html/apps/cadviewer/converter//converters/ax2024/linux/"  -xpath="/mnt/ncdata/abystrov/files/"  

try run it mannualy with bash or php, get errors 
![image](https://github.com/CADViewer/NextCloud/assets/42839959/1d49ef1d-952b-4572-ab70-350ea21a89ba)
I've tried the full command, but the result is the same.

**Comment by CADViewer** [comment]
Hello, @lehakatana ,  
1) check the content of your /apps/cadviewer/converter/converters/ax2024/linux folder.     ls -l 
2) check if the converter is alive:   ./ax2023_L64_23_12_130 -?
3) check if the converter ax2023_L64_23_12_130 has chmod 755. 

let me know your results, thanks!

**Comment by limimiking** [comment]
> Hello, @lehakatana ,
> 
> 1. check the content of your /apps/cadviewer/converter/converters/ax2024/linux folder.     ls -l
> 2. check if the converter is alive:   ./ax2023_L64_23_12_130 -?
> 3. check if the converter ax2023_L64_23_12_130 has chmod 755.
> 
> let me know your results, thanks!

Same issue, here is the output of the command:
<img width="836" alt="Screenshot 2024-07-06 at 15 32 48" src="https://github.com/CADViewer/NextCloud/assets/16589381/469dddf6-c786-444d-b3e6-b771b4ad835a">

**Comment by limimiking** [comment]
Here's part of the API Conversion log:

 exec return1  127   
output log:  
exec return1 check  127   
Q1 contentresponse  stream   
Q2:svg_creation   
Q4: $general_response{"completedAction":"svg_creation","errorCode":"E127","converter":"AutoXchange AX2020","version":"V1.00","userLabel":"fromCADViewerJS","contentLocation":"/mnt/ncdata/__groupfolders/4/陈旭/4轴机械臂固定支座.DWG","contentResponse":"stream","contentStreamData":"https://cloud.ylabics.top/apps/cadviewer/converter//php//getFile_09.php?remainOnServer=1&fileTag=f1600966834&Type=svgz"}   
Q5: 0  1  $json_response"{\"completedAction\":\"svg_creation\",\"errorCode\":\"E127\",\"converter\":\"AutoXchange AX2020\",\"version\":\"V1.00\",\"userLabel\":\"fromCADViewerJS\",\"contentLocation\":\"\/mnt\/ncdata\/__groupfolders\/4\/\u9648\u65ed\/4\u8f74\u673a\u68b0\u81c2\u56fa\u5b9a\u652f\u5ea7.DWG\",\"contentResponse\":\"stream\",\"contentStreamData\":\"https:\/\/cloud.ylabics.top\/apps\/cadviewer\/converter\/\/php\/\/getFile_09.php?remainOnServer=1&fileTag=f1600966834&Type=svgz\"}"   
$json_response = "{\"completedAction\":\"svg_creation\",\"errorCode\":\"E127\",\"converter\":\"AutoXchange AX2020\",\"version\":\"V1.00\",\"userLabel\":\"fromCADViewerJS\",\"contentLocation\":\"\/mnt\/ncdata\/__groupfolders\/4\/\u9648\u65ed\/4\u8f74\u673a\u68b0\u81c2\u56fa\u5b9a\u652f\u5ea7.DWG\",\"contentResponse\":\"stream\",\"contentStreamData\":\"https:\/\/cloud.ylabics.top\/apps\/cadviewer\/converter\/\/php\/\/getFile_09.php?remainOnServer=1&fileTag=f1600966834&Type=svgz\"}"  
$post_request_flag=1 $json_response only  
LAST BEFORE CALLBACK

**Comment by limimiking** [comment]
FYI, Nextcloud AIO create a Nextcloud container of Alpine Linux, maybe that's the reason why the converter cannot be executed.

**Comment by CADViewer** [comment]
Yes @limimiking , Alpine Linux is notorius difficult as a number of libraries are stripped out.  Can you try run a sample conversion from the folder ./ax2023_L64_23_12_130 -i=test.dwg -o=test01.svg -f=svg -model -trace  , and see if that gives a result. If it cannot run -? then it probably will not run.  Can you change to a standard Ubunu container?

**Comment by limimiking** [comment]
@CADViewer not possible to change the container because it's generated by NextCloud AIO master container. Even if I somehow change it, it will be redeployed after an auto upgrade. It means if you want to use NextCloud AIO, you have to use Alphine Linux for now.
I'll try to compile this package on Alphine Linux but I'm not sure it will work, or I hope you can compile a package for Alpine Linux as NextCloud AIO is very convenient for operation and maintenance.

**Comment by limimiking** [comment]
@CADViewer I can only compile the nodejs package. The backend converter still doesn't work. NextCloud official docker images are switching to Alpine Linux now.

**Comment by CADViewer** [comment]
Hello @limimiking , yes, please follow the Readme with the patch to add the proper libraries to Alpine Linux to run c++ executables (CADViewer back-end)
